mapper/orm/src/main/java/org/hibernate/search/mapper/orm/session/impl/HibernateOrmSearchSession.java

```diff
@@ -249,27 +245,23 @@ public PojoRuntimeIntrospector runtimeIntrospector() {
 	}
 
 	@Override
-	@SuppressWarnings("unchecked")
 	public PojoIndexingPlan<EntityReference> currentIndexingPlan(boolean createIfDoesNotExist) {
-		checkOrmSessionIsOpen();
-		Transaction transactionIdentifier = null;
-
-		TransientReference<Map<Transaction, PojoIndexingPlan<EntityReference>>> reference =
-				(TransientReference<Map<Transaction, PojoIndexingPlan<EntityReference>>>) sessionImplementor.getProperties()
-				.get( INDEXING_PLAN_PER_TRANSACTION_MAP_KEY );
-		Map<Transaction, PojoIndexingPlan<EntityReference>> planPerTransaction = reference == null ? null : reference.get();
-		if ( planPerTransaction == null ) {
-			planPerTransaction = new HashMap<>();
-			reference = new TransientReference<>( planPerTransaction );
-			sessionImplementor.setProperty( INDEXING_PLAN_PER_TRANSACTION_MAP_KEY, reference );
+		HibernateOrmSearchSessionHolder properties =
```

Comment:
This is not critical, but I think that `properties` is not a very clear name for the holder instance.

Reply:
Right, will do